django/db/models/fields/related_descriptors.py

```diff
@@ -287,11 +298,22 @@ def is_cached(self, instance):
         return hasattr(instance, self.cache_name)
 
     def get_queryset(self, **hints):
-        manager = self.related.related_model._default_manager
-        # If the related manager indicates that it should be used for
-        # related fields, respect that.
-        if not getattr(manager, 'use_for_related_fields', False):
-            manager = self.related.related_model._base_manager
+        related_model = self.related.related_model
+
+        if related_model._related_manager:
+            manager = related_model._related_manager
+        elif getattr(related_model._default_manager, 'use_for_related_fields', False):
+            warnings.warn(
+                "use_for_related_fields is deprecated use Meta.related_manager_name"
```

Comment:
same comments as above. any way we can deduplicate this "check for related manager" code?